backend/src/api/blocks.ts

```diff
@@ -343,8 +343,11 @@ class Blocks {
           await blocksRepository.$saveBlockInDatabase(blockExtended);
         }
       }
-      if (fiatConversion.ratesInitialized === true && config.DATABASE.ENABLED === true) {
-        await RatesRepository.$saveRate(blockExtended.height, fiatConversion.getConversionRates());
+      if (config.DATABASE.ENABLED === true) {
+        const blockchainInfo = await bitcoinClient.getBlockchainInfo();
+        if (blockchainInfo.blocks === blockchainInfo.headers) {
```

Comment:
We wait for Core to be synced before attempting to save block exchange rate, to avoid saving current exchange rate for old blocks (it could happen in the node backend is offline for a long time)